src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -3278,7 +3278,9 @@ private bool canGetVarArgsHandle(CORINFO_SIG_INFO* pSig)
 
         private InfoAccessType emptyStringLiteral(ref void* ppValue)
         {
-            return constructStringLiteral(_methodScope, (mdToken)CorTokenType.mdtString, ref ppValue);
+            ISymbolNode stringObject = _compilation.NodeFactory.SerializedStringObject("");
```

Comment:
Unlike `src/coreclr/tools/aot/ILCompiler.Compiler/ILCompiler.Compiler.csproj`,
`src/coreclr/tools/aot/ILCompiler.ReadyToRun/ILCompiler.ReadyToRun.csproj` and `src/coreclr/tools/aot/ILCompiler.RyuJit/ILCompiler.RyuJit.csproj`, both files are missing the partial that has `SerializedStringObject()` definition:

```xml
    <Compile Include="Compiler\DependencyAnalysis\NodeFactory.cs" />
    <!-- may also need -->
    <Compile Include="Compiler\DependencyAnalysis\NodeFactory.GenericLookups.cs" />
    <Compile Include="Compiler\DependencyAnalysis\NodeFactory.NativeLayout.cs" />
```

Reply:
Yeah thanks, turns out there was an issue in NativeAOT - `constructStringLiteral` when called by `emptyStringLiteral` could silently crash with `IndexOutOfBoundsException` and that was not reported, some how my PR made it more visible, but if you take Main and try to build `Reflection_ReflectedOnly.ilc.rsp` with first chance exceptions enabled you will see it.

`methodIL` is `ILStubMethodIL` and the method it fails on is `[S.P.CompilerGenerated]Internal.CompilerGenerated.<Module>.Format(string)`

Reply:
Yeah, NativeAOT sometimes asks RyuJIT to compile IL that doesn't come from a file on disk, but was synthesized by the compiler. RyuJIT making up a token and asking questions about it doesn't work well with that, even though it's a ECMA-spec-defined token. We could potentially fix this in `GetObject` too since the token value is defined by the spec, but this will do.